exchange/src/wrapper/algo_wrapper/binary/binary_wrapper.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "wrapper/algo_wrapper/wrapper.hpp"
+
+namespace nutc::wrapper {
+
+class BinaryWrapper : public Wrapper {
+public:
+    BinaryWrapper(
+        std::string algo, std::string trader_id, ExchangeCommunicator communicator
+    );
+
+    ~BinaryWrapper() override;
+
+private:
+    void fire_on_trade_update(
+        Ticker ticker, Side side, decimal_price price, decimal_quantity quantity
+    ) const override;
+
+    void fire_on_orderbook_update(
+        Ticker ticker, Side side, decimal_price price, decimal_quantity quantity
+    ) const override;
+
+    void fire_on_account_update(
+        Ticker ticker, Side side, decimal_price price, decimal_quantity quantity,
+        decimal_price capital
+    ) const override;
+
+    using Strategy = void;
+    using InitFunc = Strategy* (*)(MarketOrderFunction, LimitOrderFunction,
+                                   CancelOrderFunction);
+    using OnTradeUpdateFunc =
+        void (*)(Strategy*, const std::string&, const std::string&, double, double);
+    using OnOrderBookUpdateFunc = OnTradeUpdateFunc;
```

Comment:
nit: I'd rather fully define this than copy this. If two things are *coincidentally* the same, I prefer to not copy